cloud/openstack/os_port.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/python
+
+# Copyright (c) 2015 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+DOCUMENTATION = '''
+---
+module: os_port
+short_description: Add/Update/Delete ports from an OpenStack cloud.
+extends_documentation_fragment: openstack
+author: "Davide Agnello (@dagnello)"
+version_added: "2.0"
+description:
+   - Add, Update or Remove ports from an OpenStack cloud.  A state=present,
+     will ensure the port is created or updated if required.
+options:
+    network:
+     description:
+        - Network ID or name this port belongs to.
+     required: true
+   name:
+     description:
+        - Name that has to be given to the port.
+     required: false
+     default: None
+   fixed_ips:
+     description:
+        - Desired IP and/or subnet for this port.  Subnet is referenced by
+          subnet_id and IP is referenced by ip_address.
+     required: false
+     default: None
+   admin_state_up:
+     description:
+        - Sets admin state.
+     required: false
+     default: None
+   mac_address:
+     description:
+        - MAC address of this port.
+     required: false
+     default: None
+   security_groups:
+     description:
+        - Security group(s) ID(s) associated with the port (comma separated for
+          multiple security groups - no spaces between comma(s)).
+     required: false
+     default: None
+   no_security_groups:
+     description:
+        - Do not associate a security group with this port.
+     required: false
+     default: None
```

Comment:
You need to explicity define no_security_groups as a bool type. Since you are allowing it to be a string, any string value will be interpreted as true with the logic you have in this module.